spring/src/main/java/reactor/blockhound/spring/BlockHoundRunListener.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2019-Present Pivotal Software Inc, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.blockhound.spring;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.SpringApplicationRunListener;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.core.Ordered;
+import org.springframework.core.env.ConfigurableEnvironment;
+import reactor.BlockHound;
+import reactor.blockhound.integration.BlockHoundIntegration;
+
+/**
+ * This {@link SpringApplicationRunListener} installs the agent when an app is starting
+ * by calling {@link BlockHound#install(BlockHoundIntegration...)}.
+ */
+public class BlockHoundRunListener implements SpringApplicationRunListener, Ordered {
```

Comment:
@philwebb does it have to be public, or I can make it internal?

Reply:
@philwebb, has the Boot team considered making the methods in `SpringApplicationRunListener` interface `default` methods now that Java 8 is the baseline?

Reply:
Good idea, will add an issue. That API is somewhat unusual as it mainly exists to break a package tangle.

Reply:
> Good idea, will add an issue. 

Thanks

Reply:
@philwebb anything to say about the visibility of this class? 😊